### ISSUE TITLE
Highlight in decompiler view

### ISSUE THREAD

**Issue by chq-matteo** [0ffffffffh/dragondance#7]
Is it possible to highlight the corresponding source code lines in the decompiler view?

**Comment by 0ffffffffh** [comment]
it isn't at the moment. actually that's one of my to-do features that I want to implement to the dragondance. https://github.com/0ffffffffh/dragondance#the-features-that-i-want-to-bring-on-it

but i'm not able to do that because the Ghidra does not providing any pseudo code painting API. but I'm trying to find a workaround to achieve this feature. I'm looking at the ghidra source to get some hints for reliable pseudo code paint tricks when I have some free time. 

I'm going to mark this issue as a requested feature.

**Comment by chq-matteo** [comment]
Yeah, I think that using selection is the only way to highlight at all in the decompiler since it does not support markers.

Maybe it could still be a meaningful improvement?

**Comment by 0ffffffffh** [comment]
I'm not sure for now but I can say that it would not fit what we want to get. but it's underlying mechanism might be helpful to find out some way. I'm aware of that. thanks.

**Comment by kd5bwt** [comment]
I think selection would be a great improvement but it obviously depends on your goals. Using the scripting API it is pretty easy to make a selection from all code that is colored by Dragon Dance. In this snippet you need to click on the listing where it is colored and then you can run this code to make a selection matching that background color:

`from ghidra.app.plugin.core.colorizer import ColorizingService`
`service = state.getTool().getService(ColorizingService)`
`ranges = service.getBackgroundColorAddresses(service.getBackgroundColor( currentAddress ) )`
`setCurrentSelection (ranges)`
